src/components/views/rooms/RoomList.js

```diff
@@ -574,12 +574,29 @@ module.exports = React.createClass({
     },
 
     _applySearchFilter: function(list, filter) {
-        if (filter === "") return list;
+        if (filter === "") return {count: list.length};
         const lcFilter = filter.toLowerCase();
         // case insensitive if room name includes filter,
         // or if starts with `#` and one of room's aliases starts with filter
-        return list.filter((room) => (room.name && room.name.toLowerCase().includes(lcFilter)) ||
-            (filter[0] === '#' && room.getAliases().some((alias) => alias.toLowerCase().startsWith(lcFilter))));
+        let count = 0;
+        const filteredFlags = list.map((room) => {
+            const matches = (
+                    room.name &&
+                    room.name.toLowerCase().includes(lcFilter)
+                ) || (
+                    filter[0] === '#' &&
+                    room.getAliases().some((alias) => {
+                        return alias.toLowerCase().startsWith(lcFilter);
+                    })
+                );
```

Comment:
This variable is hard to read: I'd recommend breaking it out into multiple other variables and doing an OR on those.

(it was also hard to read before, and could probably use some additional variables as well)